pkg/git/repo.go

```diff
@@ -85,6 +92,63 @@ func (r *Repo) Clone(ctx context.Context) error {
 	return nil
 }
 
+func (r *Repo) ShallowClone(ctx context.Context) error {
```

Comment:
let's make this internal only, since we're only using it inside this package:

```suggestion
func (r *Repo) shallowClone(ctx context.Context) error {
```